### PR TITLE
support for streaming stats

### PR DESCRIPTION
**Pull request by ehazlett**
This adds support for streaming stats using the new Docker stats API.

Example in `examples/stats`.

### CODE REVIEW

**Comment by ehazlett** [comment]
PTAL @aluzzardi @samalba

**Comment by vieux** [comment]
@ehazlett please rebase, thanks

**Comment by ehazlett** [comment]
@vieux sorry -- done, thx!

**Comment by thaJeztah** [comment]
@ehazlett @vieux just a friendly "heads up" that there's currently a PR up that might bring an API change to the stats API; https://github.com/docker/docker/pull/10766

It hasn't been merged or approved yet, but you might want to follow progress in case it gets merged.

**Comment by vieux** [comment]
@aluzzardi what do you think ? I think we can merge this now, and update later if needed.

**Comment by ehazlett** [comment]
@thaJeztah thx -- i think this would still be beneficial and we could add the optional streaming arg if that gets merged.

**Comment by thaJeztah** [comment]
@ehazlett :+1: absolutely. I just wanted to give the heads up (in case you missed it because of work in the other repos)

**Comment by ehazlett** [comment]
@thaJeztah perfect!  thanks!

File: types.go [hunk 1]
```diff
@@ -171,3 +171,80 @@ type ImageDelete struct {
...
+	IoServicedRecursive     []BlkioStatEntry `json:"io_serviced_recursive"`
+	IoQueuedRecursive       []BlkioStatEntry `json:"io_queue_recursive"`
+	IoServiceTimeRecursive  []BlkioStatEntry `json:"io_service_time_recursive"`
+	IoWaitTimeRecursive     []BlkioStatEntry `json:"io_wait_time_recursive"`
+	IoMergedRecursive       []BlkioStatEntry `json:"io_merged_recursive"`
+	IoTimeRecursive         []BlkioStatEntry `json:"io_time_recursive"`
+	SectorsRecursive        []BlkioStatEntry `json:"sectors_recursive"`
+}
+
+type Network struct {
```
Comment (aluzzardi) [inline types.go]: This seems dangerous to be a top level struct.

How about inlining it into Stats directly (similar to ContainerInfo.State)?

**Comment by aluzzardi** [comment]
Added a minor comment, other than that LGTM

**Comment by ehazlett** [comment]
@aluzzardi updated to inline `Network`.  PTAL  thanks!

**Comment by vieux** [comment]
LGTM

**Comment by ehazlett** [comment]
Thanks!

**Comment by vieux** [comment]
@ehazlett https://github.com/samalba/dockerclient/commit/9c34339a5d71de3ef3c230c1d121ebbe16a95cb9 :(

**Pull request merged**